File: RestaurantApp/app/src/main/java/com/yarin/restaurantapp/MainActivity.java
package com.yarin.restaurantapp;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

public class MainActivity extends AppCompatActivity {

    private CardView startersCard, mainsCard, desertsCard;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        startersCard = findViewById(R.id.card_view_starters);
        mainsCard = findViewById(R.id.card_view_main);
        desertsCard = findViewById(R.id.card_view_deserts);

        startersCard.setOnClickListener(v -> startActivity(new Intent(this, StartersActivity.class)));
        mainsCard.setOnClickListener(v -> startActivity(new Intent(this, MainCoursesActivity.class)));
        desertsCard.setOnClickListener(v -> startActivity(new Intent(this, DesertsActivity.class)));

        final TextView emailView = findViewById(R.id.text_email);
        emailView.setOnClickListener(v -> {
            final Intent emailIntent = new Intent(Intent.ACTION_SENDTO);
            emailIntent.setData(Uri.parse("mailto:somemail@gmail.com"));
            startActivity(emailIntent);
        });

    }
}

File: LoginFormChallenge/LoginFormChallenge/app/src/main/java/com/yarin/loginformchallenge/MainActivity.kt
package com.yarin.loginformchallenge

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import com.yarin.loginformchallenge.databinding.ActivityMainBinding
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        setupSpinner()
        setupButton()

    }

    private fun setupButton() {
        binding.button.setOnClickListener {
            createAccount()
        }
    }

    private fun setupSpinner() {
        val titles = arrayOf("Miss", "Mrs", "Ms", "Mr", "Mx", "Dr", "Prof")
        val titleAdapter = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, titles)
        binding.spinner.adapter = titleAdapter
    }

    private fun createAccount() {
        val user = User(
            binding.spinner.selectedItem.toString(),
            binding.editTextFirstname.text.toString(),
            binding.editTextLastname.text.toString(),
            binding.editTextEmail.text.toString(),
            binding.editTextPhone.text.toString(),
            binding.editTextPassword.text.toString(),
        )

        val intent = Intent(this, SummaryActivity::class.java)
        intent.putExtra("User", user)
        startActivity(intent)
    }

}